### PR TITLE
1131 dockerification

### PR DESCRIPTION
**Pull request by OSdave**

### CODE REVIEW

File: lib/talktome/app.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require 'sinatra'
+
+module Talktome
+  class App < Sinatra::Application
+
+    post '/contact' do
```
Comment (blambeau) [inline lib/talktome/app.rb:6]: `/contact-us/`

File: lib/talktome/app.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      begin
+        body = request.body.read
+        info = JSON.parse(body)
+        raise ::Talktome::InvalidEmailError unless info['email'] && info['email'] =~ /^[^@]+@[^@]+$/
+        raise ::Talktome::InvalidEmailError if info['confirmEmail'] && info['confirmEmail'] =~ /^[^@]+@[^@]+$/
```
Comment (blambeau) [inline lib/talktome/app.rb:11]: Document the fact that this is a honeypot.

File: lib/talktome/app.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        user = {
+          email: info['email']
+        }
+        data = info['data']
+
+        TALKTOME = Talktome::Client::Local.new(Path(info['folder']['data']))
```
Comment (blambeau) [inline lib/talktome/app.rb:17]: Don't do this. Security issue.

File: spec/app/test_app.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+      it 'works' do
+        post "/contact", {
+          email: 'hello@visitor.com',
+          folder: folder,
+          template: template,
+          from: from,
+          data: { who: "Test user" }
+        }.to_json, { "CONTENT_TYPE" => "application/json" }
+        File.open('/home/david/Desktop/response.html', 'w') { |file| file.write(last_response.body)}
```
Comment (blambeau) [inline spec/app/test_app.rb:33]: Remove.

**Review by blambeau — CHANGES_REQUESTED** [review]

**Comment by blambeau** [comment]
Regarding folder, template & from:

1. Folder = convention: a `./talktome` folder must exists (typically mounted as a volume). 

Typically used as follows (Dockerfile):

```
FROM enspirit/talktome:latest
COPY ./mail-templates /app/talktome/
```

Another way (docker-compose):

```
services:
  talktome:
    image: enspirit/talktome
    volumes:
      - ./mail-templates/:/app/talktome/
```

2. Template = convention. `/contact-us` requires a `contact-us/email.md` in the templates.

3. Regarding the `FROM`. Use the `TALKTOME_EMAIL_DEFAULT_FROM` environment variables (must be set by Devops).

4.' (option) We could enhance Talktome to have a better notion of 'default_email' at configuration time, with more env variables for subject, to, from, reply-to etc. Every mail sent would then be a overriding of the default email.

**Comment by OSdave** [comment]
Jekins test this please.

**Comment by OSdave** [comment]
@blambeau let me know if you think I can merge this branch into master and go on with the integration for exquando

File: lib/talktome.rb [hunk 5]
```diff
@@ -65,3 +68,4 @@ def auto_options(folder)
 require 'talktome/strategy'
 require 'talktome/message'
 require 'talktome/client'
+require 'talktome/app'
```
Comment (blambeau) [inline lib/talktome.rb:71]: Don't require by default, move it to config.ru

**Review by blambeau — CHANGES_REQUESTED** [review]

**Pull request merged**